Exercises for Lesson03 and Lesson04

zyang2020 opened the pull request.

charlesrobison commented on students/zhen_yang/lesson03/list_lab.py:
```diff
@@ -0,0 +1,140 @@
+
+# 1.3 Ask user a number and display the number back and 
+# the fruit corresponding to that number  
+input_str = input(f"Please input a number within range: [1:{len(my_list)}] ")
```
Nice!

charlesrobison commented on students/zhen_yang/lesson03/list_lab.py:
```diff
@@ -0,0 +1,140 @@
+print(f"Add item to list using + : {my_list}.")
+
+# 1.5 Add another fruit to the beginning of the list using insert() 
+my_list.insert(0,'Avocado')
```
Missing whitespace after comma

charlesrobison commented on students/zhen_yang/lesson03/list_lab.py:
```diff
@@ -0,0 +1,140 @@
+# Iterate the total number of occurances, then get the index of the 
+# first occurence of the fruit in the list and remove it by pop()
+for i in range(my_count):
+      the_index = my_list.index(input_str)
```
Over indented

charlesrobison commented on students/zhen_yang/lesson03/list_lab.py:
```diff
@@ -0,0 +1,140 @@
+# first occurence of the fruit in the list and remove it by pop()
+for i in range(my_count):
+      the_index = my_list.index(input_str)
+      my_list.pop(the_index)
```
Here too

charlesrobison commented on students/zhen_yang/lesson03/list_lab.py:
```diff
@@ -0,0 +1,140 @@
+# Make a new list with the contents of the original, 
+# but with all the letters in each item reversed
+my_list = backup_list[:]
+new_list =[]
```
Missing whitespace after assignment operator

charlesrobison commented on students/zhen_yang/lesson03/list_lab.py:
```diff
@@ -0,0 +1,140 @@
+
+    # delete this fruit from the list if the answer from user is 'no'
+    if input_str == 'no': 
+        my_list.remove(i)
```
Works well

charlesrobison commented on students/zhen_yang/lesson03/mailroom_part1.py:
```diff
@@ -0,0 +1,223 @@
+# find the name from donors_db 
+##################
+def found_name(my_name):
+    count = 0
```
Variable created but not used

charlesrobison commented on students/zhen_yang/lesson03/mailroom_part1.py:
```diff
@@ -0,0 +1,223 @@
+# add the amount to the existing donor on the donors_db 
+##################
+def add_amount(amount,r_index):
+    tmp_amount = [amount]
```
Variable not used

charlesrobison commented on students/zhen_yang/lesson03/mailroom_part1.py:
```diff
@@ -0,0 +1,223 @@
+##################
+# add the amount to the existing donor on the donors_db 
+##################
+def add_amount(amount,r_index):
```
Missing whitespace after comma

charlesrobison commented on students/zhen_yang/lesson03/mailroom_part1.py:
```diff
@@ -0,0 +1,223 @@
+##################
+# add the new orcorder to the donors_db
+##################
+def add_newrecord(d_name,amount):
```
Missing whitespace after comma

charlesrobison commented on students/zhen_yang/lesson03/mailroom_part1.py:
```diff
@@ -0,0 +1,223 @@
+##################
+# print the thank you email 
+##################
+def thankyou_letter(d_name,amount):
```
Missing whitespace here

charlesrobison commented on students/zhen_yang/lesson03/mailroom_part1.py:
```diff
@@ -0,0 +1,223 @@
+#################################
+def quit_program():
+    print("Bye!")
+    exit()
```
This isn't actually quitting the program

charlesrobison commented on students/zhen_yang/lesson03/mailroom_part1.py:
```diff
@@ -0,0 +1,223 @@
+            tot_amount = tot_amount + j 
+        avg_amount = tot_amount / count
+        # print out current row data
+        print(formater_content.format(name,tot_amount,count,avg_amount))
```
Isn't sorting correctly for new donations

zyang2020 commented:
Hello
I fixed my lesson03 style errors, bugs and finished my lesson04 exercise. I did another PR #809. I have some questions regarding to the errors of exercise and submit.
1. Each time I fixed the errors and bugs according to your comments, I just merge these new commits to my next PR, right?
2. I found on the course webpage, it also ask submit the exercises for each lesson by uploading the files over there and copy paste the PR url over there. Do I need to do this procedure?


Thank you,

Zhen
Sent from my iPad

On Mar 27, 2020, at 8:50 PM, Charles Robison <notifications@github.com<mailto:notifications@github.com>> wrote:


@charlesrobison approved this pull request.

Lesson 3 files looking good overall. Mailroom appears to have a bug - adding new donations isn't sorting correctly. @zyang2020<https://github.com/zyang2020> - lesson 4 mailroom doesn't run. Are you ready for a review on the lesson 4 files? When complete, please submit in another PR. Please reach out if you have questions.

Ready to merge

________________________________

In students/zhen_yang/lesson03/list_lab.py<https://github.com/UWPCE-PythonCert-ClassRepos/SP_Online_PY210/pull/802#discussion_r399613581>:

> +#####################
+# Series 1
+#####################
+# 1.1 Create a list
+print("----------- Series 1 ---------")
+my_list = ['Apples','Pears','Oranges','Peaches']
+#print(f"My fruit list is: {my_list}.")
+
+# 1.2 Add item to the list
+new_item = input("Please input another fruit: ")
+my_list.append(new_item)
+#print(f"1. My updated fruit list is: {my_list}.")
+
+# 1.3 Ask user a number and display the number back and
+# the fruit corresponding to that number
+input_str = input(f"Please input a number within range: [1:{len(my_list)}] ")


Nice!

________________________________

In students/zhen_yang/lesson03/list_lab.py<https://github.com/UWPCE-PythonCert-ClassRepos/SP_Online_PY210/pull/802#discussion_r399613687>:

> +# the fruit corresponding to that number
+input_str = input(f"Please input a number within range: [1:{len(my_list)}] ")
+input_num = int(input_str)
+if input_num >= 1 and input_num <= len(my_list):
+    print(f"Input Number is: {input_num} and the corresponding fruit is: {my_list[input_num-1]}")
+else:
+    print("Input Number is out of list range!")
+    exit()
+
+# 1.4 Add another fruit to the beginning of the list using '+'
+new_fruit = ['Watermelon']
+my_list = new_fruit + my_list
+print(f"Add item to list using + : {my_list}.")
+
+# 1.5 Add another fruit to the beginning of the list using insert()
+my_list.insert(0,'Avocado')


Missing whitespace after comma

________________________________

In students/zhen_yang/lesson03/list_lab.py<https://github.com/UWPCE-PythonCert-ClassRepos/SP_Online_PY210/pull/802#discussion_r399613783>:

> +# Count the number of times that the fruit name appears in the list
+my_count = my_list.count(input_str)
+
+# if the input fruit does not appear in the list,
+# we will keep asking a name for a fruit.
+while my_count == 0:
+    input_str = input("Can't find the name from the list. Please input another name: ")
+    my_count = my_list.count(input_str)
+#print(f"Found the fruit:{input_str} appears in the list {my_count} times.")
+
+
+# Found the fruit from the list, then we delete all the occurrences
+# Iterate the total number of occurances, then get the index of the
+# first occurence of the fruit in the list and remove it by pop()
+for i in range(my_count):
+      the_index = my_list.index(input_str)


Over indented

________________________________

In students/zhen_yang/lesson03/list_lab.py<https://github.com/UWPCE-PythonCert-ClassRepos/SP_Online_PY210/pull/802#discussion_r399613788>:

> +my_count = my_list.count(input_str)
+
+# if the input fruit does not appear in the list,
+# we will keep asking a name for a fruit.
+while my_count == 0:
+    input_str = input("Can't find the name from the list. Please input another name: ")
+    my_count = my_list.count(input_str)
+#print(f"Found the fruit:{input_str} appears in the list {my_count} times.")
+
+
+# Found the fruit from the list, then we delete all the occurrences
+# Iterate the total number of occurances, then get the index of the
+# first occurence of the fruit in the list and remove it by pop()
+for i in range(my_count):
+      the_index = my_list.index(input_str)
+      my_list.pop(the_index)


Here too

________________________________

In students/zhen_yang/lesson03/list_lab.py<https://github.com/UWPCE-PythonCert-ClassRepos/SP_Online_PY210/pull/802#discussion_r399613838>:

> +
+    # delete this fruit from the list if the answer from user is 'no'
+    if input_str == 'no':
+        my_list.remove(i)
+
+print(f"The final favoriate fruit list: {my_list}")
+
+
+
+#####################
+# Series 4
+#####################
+# Make a new list with the contents of the original,
+# but with all the letters in each item reversed
+my_list = backup_list[:]
+new_list =[]


Missing whitespace after assignment operator

________________________________

In students/zhen_yang/lesson03/list_lab.py<https://github.com/UWPCE-PythonCert-ClassRepos/SP_Online_PY210/pull/802#discussion_r399613857>:

> +#####################
+# Series 3
+#####################
+my_list = backup_list[:]
+# Ask the user "Do you like apples?"
+
+print("----------- Series 3 ---------")
+# !!! iterate over a copy and mutate the original list
+for i in my_list[:]:
+    input_str = input(f"Do you like {i.lower()}?: ")
+    while input_str != 'yes' and input_str != 'no':
+        input_str = input("Please input 'yes' or 'no': ")
+
+    # delete this fruit from the list if the answer from user is 'no'
+    if input_str == 'no':
+        my_list.remove(i)


Works well

________________________________

In students/zhen_yang/lesson03/mailroom_part1.py<https://github.com/UWPCE-PythonCert-ClassRepos/SP_Online_PY210/pull/802#discussion_r399614080>:

> +           return -1
+
+    if  float(input_str)>=0:
+        input_amount = float(input_str)
+        return input_amount
+    else:# input is not a number we need to ask user to input number again.
+        print("Plese input a positive number for donation amount. Thank you! ")
+        return -1
+
+
+
+##################
+# find the name from donors_db
+##################
+def found_name(my_name):
+    count = 0


Variable created but not used

________________________________

In students/zhen_yang/lesson03/mailroom_part1.py<https://github.com/UWPCE-PythonCert-ClassRepos/SP_Online_PY210/pull/802#discussion_r399614095>:

> +
+##################
+# find the name from donors_db
+##################
+def found_name(my_name):
+    count = 0
+    for rowindex, row in enumerate(donors_db):
+        if row[0] == my_name:
+            return rowindex
+    return -1
+
+##################
+# add the amount to the existing donor on the donors_db
+##################
+def add_amount(amount,r_index):
+    tmp_amount = [amount]


Variable not used

________________________________

In students/zhen_yang/lesson03/mailroom_part1.py<https://github.com/UWPCE-PythonCert-ClassRepos/SP_Online_PY210/pull/802#discussion_r399614127>:

> +
+
+##################
+# find the name from donors_db
+##################
+def found_name(my_name):
+    count = 0
+    for rowindex, row in enumerate(donors_db):
+        if row[0] == my_name:
+            return rowindex
+    return -1
+
+##################
+# add the amount to the existing donor on the donors_db
+##################
+def add_amount(amount,r_index):


Missing whitespace after comma

________________________________

In students/zhen_yang/lesson03/mailroom_part1.py<https://github.com/UWPCE-PythonCert-ClassRepos/SP_Online_PY210/pull/802#discussion_r399614143>:

> +            return rowindex
+    return -1
+
+##################
+# add the amount to the existing donor on the donors_db
+##################
+def add_amount(amount,r_index):
+    tmp_amount = [amount]
+    donors_db[r_index][1].append(amount)
+    print(f"3. Updated donor amount: {donors_db[r_index]}")
+
+
+##################
+# add the new orcorder to the donors_db
+##################
+def add_newrecord(d_name,amount):


Missing whitespace after comma

________________________________

In students/zhen_yang/lesson03/mailroom_part1.py<https://github.com/UWPCE-PythonCert-ClassRepos/SP_Online_PY210/pull/802#discussion_r399614160>:

> +
+##################
+# add the new orcorder to the donors_db
+##################
+def add_newrecord(d_name,amount):
+    # use tuple to make sure that a single record only can
+    # contain two items: "name" and "donation amount"
+    new_record = (d_name,[amount])
+    donors_db.append(new_record)
+    #print(f"1. Updated record {donors_db}")
+
+
+##################
+# print the thank you email
+##################
+def thankyou_letter(d_name,amount):


Missing whitespace here

________________________________

In students/zhen_yang/lesson03/mailroom_part1.py<https://github.com/UWPCE-PythonCert-ClassRepos/SP_Online_PY210/pull/802#discussion_r399614263>:

> +
+    print("\n")
+
+
+
+
+
+
+
+
+#################################
+# define Quit_Program()  Option #
+#################################
+def quit_program():
+    print("Bye!")
+    exit()


This isn't actually quitting the program

________________________________

In students/zhen_yang/lesson03/mailroom_part1.py<https://github.com/UWPCE-PythonCert-ClassRepos/SP_Online_PY210/pull/802#discussion_r399614278>:

> +
+    # print the Title of the report
+    print(formater_title.format(col_1,col_2,col_3,col_4))
+    print('-'*71)
+
+    # print the content of the report
+    for i in donors_db:
+        tot_amount = 0
+        avg_amount = 0
+        name = i[0]
+        count = len(i[1])
+        for j in i[1]:
+            tot_amount = tot_amount + j
+        avg_amount = tot_amount / count
+        # print out current row data
+        print(formater_content.format(name,tot_amount,count,avg_amount))


Isn't sorting correctly for new donations

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/UWPCE-PythonCert-ClassRepos/SP_Online_PY210/pull/802#pullrequestreview-383279397>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AO4WN3MK5ANVUZUQUH7YJWLRJVXY3ANCNFSM4LTXXAGA>.

natasha-aleksandrova commented:
@zyang2020 

1. yes you would create a new PR for your fixes, unless one is already open
2. yes you will need to upload the files there as well, per UW policy.

Thanks!
